tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,415 @@
+            ],
+            [
+                'key' => 'has_key_test_1',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            true,
+        ];
+        // simple test in group context, when value is set for this context
+        yield [
+            [
+                'key' => 'has_key_test_2',
+                'value' => 'has_value_test_2',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            [
+                'key' => 'has_key_test_2',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            true,
+        ];
+        // simple test in single shop context, when value is set for this context
+        yield [
+            [
+                'key' => 'has_key_test_3',
+                'value' => 'has_value_test_3',
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            [
+                'key' => 'has_key_test_3',
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            true,
+        ];
+        // test in group context, value is set for all shop context (not strict)
+        yield [
+            [
+                'key' => 'has_key_test_4',
+                'value' => 'has_value_test_4',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            [
+                'key' => 'has_key_test_4',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            true,
+        ];
+        // test in single shop context, value is set for parent group context (not strict)
+        yield [
+            [
+                'key' => 'has_key_test_5',
+                'value' => 'has_value_test_5',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            [
+                'key' => 'has_key_test_5',
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            true,
+        ];
+        // test in group context with $isStrict = true, when value is set for all shop context
+        yield [
+            [
+                'key' => 'has_key_test_6',
+                'value' => 'has_value_test_6',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            [
+                'key' => 'has_key_test_6',
+                'shopConstraint' => ShopConstraint::shopGroup(1, true),
+            ],
+            false,
+        ];
+        // test in single shop context with $isStrict = true, when value is set for parent group
+        yield [
+            [
+                'key' => 'has_key_test_7',
+                'value' => 'has_value_test_7',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            [
+                'key' => 'has_key_test_7',
+                'shopConstraint' => ShopConstraint::shopGroup(2, true),
```

Comment:
```suggestion
                'shopConstraint' => ShopConstraint::shop(2, true),
```

I think there's a small typo here, since you say in the comment you wat to test single shop context from group value